### PR TITLE
[#3752] Distinguish domains we're pretty sure are spam from mere suspicious ones

### LINKED ISSUE #3752: Add new spam scorer domains
Since banning over 1000 users from WDTK yesterday, about 20 more from the same domains have appeared so we should consider either banning new signups outright or at least weighting them as part of the spam scorer process.

(We also have a new signup from a webgarden.com address having decided to purge all of the current signups from there, a less pressing problem but probably worth noting.)

<details><summary>Affected domains</summary><p>
allemaling.com  <br />
brmailing.com  <br />
checknowmail.com  <br />
colde-mail.com  <br />
consimail.com  <br />
continumail.com  <br />
contumail.com  <br />
emailber.com  <br />
grow-mail.com  <br />
inemaling.com  <br />
inmailing.com  <br />
itemailing.com  <br />
itmailing.com  <br />
kod-emailing.com  <br />
kod-maling.com  <br />
kodemailing.com  <br />
kodmailing.com  <br />
left-mail.com  <br />
mabermail.com   <br />
mailphar.com  <br />
out-email.com  <br />
semi-mile.com  <br />
sin-mailing.com  <br />
sinemailing.com  <br />
sinmailing.com  <br />
takmailing.com  <br />
themailemail.com  <br />
visitinbox.com  <br />
wowmailing.com  <br />
</p></details>

### PR DESCRIPTION
**Pull request by lizconlan**
...and prevent new user signups from spam domains.

Closes #3752

### CODE REVIEW

File: lib/user_spam_scorer.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class UserSpamScorer
     :name_includes_non_alpha_characters? => 3,
     :name_is_garbled? => 5,
     :email_from_suspicious_domain? => 5,
+    :email_from_spam_domain? => 8,
```
Comment (lizconlan) [inline lib/user_spam_scorer.rb]: 8 seemed like a good number but open to suggestions

File: app/controllers/user_controller.rb [hunk 2]
```diff
@@ -185,6 +185,25 @@ def signup
...
+          if send_exception_notifications?
+            msg = "Attempted signup from spam domain email: " \
+                  " #{ @user_signup.email }"
+            e = Exception.new(msg)
+            ExceptionNotifier.notify_exception(e, :env => request.env)
+          end
+
+          if AlaveteliConfiguration.enable_anti_spam
+            flash.now[:error] =
+              _("Sorry, we're currently unable to sign up new users, " \
```
Comment (lizconlan) [inline app/controllers/user_controller.rb]: I'm not convinced this is the correct text - should we be telling people "we don't accept email from **webexample.co.uk**" or just the same generic text as the IP block?

Comment (lizconlan) [inline app/controllers/user_controller.rb]: Suggestions welcome

File: app/controllers/user_controller.rb [hunk 3]
```diff
@@ -185,6 +185,25 @@ def signup
           end
         end
 
+        # Prevent signups from spam domains
+        if UserSpamScorer.new.email_from_spam_domain?(@user_signup)
+          if send_exception_notifications?
+            msg = "Attempted signup from spam domain email: " \
+                  " #{ @user_signup.email }"
```
Comment (garethrees) [inline app/controllers/user_controller.rb]: Don't need the space at the start of this line given the space at the end of the previous line.

File: app/controllers/user_controller.rb (hunk 2 above)
Comment (garethrees) [inline app/controllers/user_controller.rb]: I think we should keep it as-is for now. We haven't got anti-spam on for most sites, so no one will really see this unless we're dealing with an influx.

**Comment by lizconlan** [comment]
@garethrees good spot on the typo, thanks!

**Pull request merged**